Prosedur yang akan saya lakukan untuk menyelesaikan proyek ini yaitu:

- Persiapan Data:

Mengunduh dataset dari tautan yang diberikan.
Melakukan pembersihan data jika diperlukan, seperti menangani nilai yang hilang atau tidak valid.
Melakukan pengecekan statistik deskriptif untuk memahami distribusi data.

- Pelatihan dan Evaluasi Model:

Membagi data menjadi training set dan validation set dengan rasio 75:25.
Melatih model regresi linear untuk setiap wilayah menggunakan training set.
Membuat prediksi untuk validation set dan menyimpan hasilnya.
Menghitung rata-rata volume pemesanan dan Root Mean Squared Error (RMSE) dari model untuk setiap wilayah.
Menganalisis hasilnya untuk memilih wilayah terbaik.

- Perhitungan Laba:

Simpan nilai yang dibutuhkan untuk perhitungan laba.
Membandingkan rata-rata volume pemesanan dengan target volume untuk menghindari kerugian.
Memberikan kesimpulan tentang persiapan yang dilakukan dalam langkah perhitungan laba.

- Fungsi Perhitungan Laba:

Membuat fungsi untuk menghitung keuntungan dari 200 sumur terpilih.
Simpan prediksi untuk 200 sumur di setiap wilayah.
Memberikan kesimpulan tentang wilayah yang cocok untuk pengembangan sumur minyak berdasarkan keuntungan yang dihitung.

- Risiko dan Keuntungan:

Menggunakan teknik bootstrapping dengan 1.000 sampel untuk menemukan distribusi profit.
Menghitung rata-rata profit, interval kepercayaan 95%, dan risiko kerugian.
Memberikan kesimpulan tentang wilayah yang cocok untuk pengembangan sumur minyak berdasarkan analisis risiko dan keuntungan.

Langkah-langkah ini akan membantu saya dalam mempersiapkan, melatih, dan menganalisis model untuk memilih wilayah terbaik untuk mengembangkan sumur minyak baru dengan mempertimbangkan keuntungan dan risiko.








In [1]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
# Memuat dataset geo_data_0.csv
geo_data_0 = pd.read_csv('/datasets/geo_data_0.csv')

# Eksplorasi Data
# Melihat informasi umum tentang dataset
print(geo_data_0.info())

# Melihat sample data
print(geo_data_0.sample(5))

# Statistik deskriptif
print(geo_data_0.describe())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None
          id        f0        f1        f2     product
84359  AKyCQ -0.057470  0.292567  6.864681  129.260555
95841  zN7x4  0.547547 -0.275073  3.076626  100.343969
15692  cJndF -0.777675  0.357170  1.725882    9.456085
56144  xjBIZ  1.824029  0.058676  2.378386  129.959176
59920  JThAr -0.493893  0.987818  7.885112   86.463372
                  f0             f1             f2        product
count  100000.000000  100000.000000  100000.000000  100000.000000
mean        0.500419       0.250143       2.502647      92.500000
std         0.871832       0.5044

Hasil tersebut memberikan informasi tentang dataset geo_data_0.csv:

- Informasi Umum Dataset:

Dataset ini memiliki 100.000 baris dan 5 kolom.
Tidak ada nilai yang hilang (non-null count untuk setiap kolom adalah 100.000).
Tipe data dari kolom id adalah object, sedangkan tipe data dari kolom f0, f1, f2, dan product adalah float64.
Memory usage dataset ini adalah sekitar 3.8+ MB.

- Sample Data:

Ditampilkan lima baris acak dari dataset, yang menunjukkan struktur data yang dimiliki.

- Statistik Deskriptif:

Statistik deskriptif seperti rata-rata, standar deviasi, nilai minimum, kuartil, dan nilai maksimum dari setiap kolom numerik ditampilkan.
Kolom 'product' memiliki rata-rata sekitar 92.5 dan standar deviasi sekitar 44.29, dengan nilai minimum 0 dan maksimum 185.36.
Kolom 'f0', 'f1', dan 'f2' juga memiliki statistik deskriptif yang ditampilkan. Misalnya, rata-rata untuk 'f0' adalah sekitar 0.500, 'f1' sekitar 0.250, dan 'f2' sekitar 2.503.

Informasi ini memberikan gambaran awal tentang struktur dan distribusi data dalam dataset geo_data_0.csv.








In [3]:
# Memuat dataset geo_data_1.csv
geo_data_1 = pd.read_csv('/datasets/geo_data_1.csv')

# Eksplorasi Data
# Melihat informasi umum tentang dataset
print(geo_data_1.info())

# Melihat sample data
print(geo_data_1.sample(5))

# Statistik deskriptif
print(geo_data_1.describe())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None
          id         f0        f1        f2     product
23666  RAO2T  -5.125802 -5.121863 -0.003303    3.179103
49940  Inix7  -3.733356 -3.836883  1.995977   57.085625
3770   WMSQO  -3.940061  2.434423  4.994574  137.945408
91389  YqF6Z -12.589518 -0.419388  0.994012   30.132364
51347  CXRO5   9.195325 -4.569706  2.011481   53.906522
                  f0             f1             f2        product
count  100000.000000  100000.000000  100000.000000  100000.000000
mean        1.141296      -4.796579       2.494541      68.825000
std         8.965932       


Hasil tersebut memberikan informasi tentang dataset geo_data_1.csv:

- Informasi Umum Dataset:

Dataset ini juga memiliki 100.000 baris dan 5 kolom, sama dengan geo_data_0.csv.
Tidak ada nilai yang hilang (non-null count untuk setiap kolom adalah 100.000).
Tipe data dari kolom id adalah object, sedangkan tipe data dari kolom f0, f1, f2, dan product adalah float64.
Memory usage dataset ini juga adalah sekitar 3.8+ MB, sama dengan geo_data_0.csv.

- Sample Data:

Ditampilkan lima baris acak dari dataset, yang memberikan gambaran tentang struktur data yang dimiliki.

- Statistik Deskriptif:

Statistik deskriptif seperti rata-rata, standar deviasi, nilai minimum, kuartil, dan nilai maksimum dari setiap kolom numerik ditampilkan.
Kolom 'product' memiliki rata-rata sekitar 68.83 dan standar deviasi sekitar 45.94, dengan nilai minimum 0 dan maksimum 137.95.
Kolom 'f0', 'f1', dan 'f2' juga memiliki statistik deskriptif yang ditampilkan. Misalnya, rata-rata untuk 'f0' adalah sekitar 1.14, 'f1' adalah sekitar -4.80, dan 'f2' adalah sekitar 2.49.

Informasi ini memberikan gambaran awal tentang struktur dan distribusi data dalam dataset geo_data_1.csv, yang mirip dengan geo_data_0.csv.

In [4]:
# Memuat dataset geo_data_2.csv
geo_data_2 = pd.read_csv('/datasets/geo_data_2.csv')

# Eksplorasi Data
# Melihat informasi umum tentang dataset
print(geo_data_2.info())

# Melihat sample data
print(geo_data_2.sample(5))

# Statistik deskriptif
print(geo_data_2.describe())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB
None
          id        f0        f1        f2     product
53474  1k18o  0.343663 -0.081959  5.360197  146.655892
83255  aZkhb -0.871845  0.464063  3.916263  130.504636
18208  YBJdN -1.265130  3.018684  4.429033   98.596590
30074  ZPPUp -3.253399 -2.218321 -3.622253   53.114992
72769  CKXug -0.726212  0.433624  1.571383    8.182555
                  f0             f1             f2        product
count  100000.000000  100000.000000  100000.000000  100000.000000
mean        0.002023      -0.002081       2.495128      95.000000
std         1.732045       1.7304

Hasil tersebut memberikan informasi tentang dataset geo_data_2.csv:

- Informasi Umum Dataset:

Dataset ini juga memiliki 100.000 baris dan 5 kolom, seperti geo_data_0.csv dan geo_data_1.csv.
Tidak ada nilai yang hilang (non-null count untuk setiap kolom adalah 100.000).
Tipe data dari kolom id adalah object, sedangkan tipe data dari kolom f0, f1, f2, dan product adalah float64.
Memory usage dataset ini juga adalah sekitar 3.8+ MB, sama dengan dua dataset sebelumnya.

- Sample Data:

Ditampilkan lima baris acak dari dataset, yang memberikan gambaran tentang struktur data yang dimiliki.

- Statistik Deskriptif:

Statistik deskriptif seperti rata-rata, standar deviasi, nilai minimum, kuartil, dan nilai maksimum dari setiap kolom numerik ditampilkan.
Kolom 'product' memiliki rata-rata sekitar 95.00 dan standar deviasi sekitar 44.75, dengan nilai minimum 0 dan maksimum 190.03.
Kolom 'f0', 'f1', dan 'f2' juga memiliki statistik deskriptif yang ditampilkan. Misalnya, rata-rata untuk 'f0' adalah sekitar 0.002, 'f1' adalah sekitar -0.002, dan 'f2' adalah sekitar 2.495.

Informasi ini memberikan gambaran awal tentang struktur dan distribusi data dalam dataset geo_data_2.csv, yang serupa dengan dua dataset sebelumnya.








In [5]:
# Melatih dan menguji model untuk setiap wilayah di 'geo_data_0.csv':

# Pisahkan fitur dan target
X = geo_data_0[['f0', 'f1', 'f2']]
y = geo_data_0['product']

# Pisahkan data menjadi training set dan validation set
X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.25, random_state=42)

# Inisialisasi model regresi linear
model = LinearRegression()

# Latih model menggunakan data training
model.fit(X_train, y_train)

# Buat prediksi untuk data validation
predictions = model.predict(X_val)

# Simpan prediksi dan jawaban yang benar untuk validation set
results_df = pd.DataFrame({'Actual': y_val, 'Predicted': predictions})

# Tampilkan prediksi volume rata-rata pemesanan
print("Rata-rata Prediksi Volume Pemesanan:", predictions.mean())

# Hitung RMSE (Root Mean Squared Error)
rmse = np.sqrt(mean_squared_error(y_val, predictions))
print("RMSE dari Model:", rmse)

Rata-rata Prediksi Volume Pemesanan: 92.3987999065777
RMSE dari Model: 37.756600350261685


Hasil dari pelatihan dan pengujian model untuk wilayah di 'geo_data_0.csv' adalah sebagai berikut:

- Rata-rata Prediksi Volume Pemesanan: 92.40 ribu barel (dibulatkan).

Ini adalah nilai rata-rata dari prediksi volume pemesanan yang dihasilkan oleh model untuk data validasi.

- RMSE (Root Mean Squared Error): 37.76 ribu barel.

RMSE mengukur seberapa baik model kita cocok dengan data yang diamati. Nilai RMSE yang lebih rendah menunjukkan bahwa model memiliki kinerja yang lebih baik.
Dalam konteks ini, RMSE yang cukup besar menunjukkan bahwa model mungkin tidak sesuai dengan data sebaik yang diharapkan.

- Analisis:

1. Rata-rata prediksi volume pemesanan cukup dekat dengan nilai sebenarnya dari data validasi, yang menunjukkan bahwa model mungkin cukup baik dalam melakukan prediksi secara keseluruhan.
2. Namun, RMSE yang cukup tinggi (lebih dari 25% dari rata-rata nilai target) menunjukkan bahwa model mungkin tidak cukup akurat dalam memprediksi volume cadangan di sumur minyak baru.
3. Kemungkinan terdapat faktor-faktor lain yang tidak dimasukkan ke dalam model dan memengaruhi hasil prediksi, atau model yang digunakan mungkin perlu penyesuaian lebih lanjut untuk meningkatkan kinerjanya.

Dengan demikian, hasil tersebut memberikan wawasan awal tentang kinerja model untuk wilayah di 'geo_data_0.csv', namun perlu dilakukan analisis lebih lanjut dan penyesuaian model jika diperlukan untuk meningkatkan akurasinya.








In [6]:
# Melatih dan menguji model untuk setiap wilayah di 'geo_data_1.csv':

# Pisahkan fitur dan target
X = geo_data_1[['f0', 'f1', 'f2']]
y = geo_data_1['product']

# Pisahkan data menjadi training set dan validation set
X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.25, random_state=42)

# Inisialisasi model regresi linear
model = LinearRegression()

# Latih model menggunakan data training
model.fit(X_train, y_train)

# Buat prediksi untuk data validation
predictions = model.predict(X_val)

# Simpan prediksi dan jawaban yang benar untuk validation set
results_df = pd.DataFrame({'Actual': y_val, 'Predicted': predictions})

# Tampilkan prediksi volume rata-rata pemesanan
print("Rata-rata Prediksi Volume Pemesanan:", predictions.mean())

# Hitung RMSE (Root Mean Squared Error)
rmse = np.sqrt(mean_squared_error(y_val, predictions))
print("RMSE dari Model:", rmse)

Rata-rata Prediksi Volume Pemesanan: 68.71287803913762
RMSE dari Model: 0.890280100102884


Hasil dari pelatihan dan pengujian model untuk wilayah di 'geo_data_1.csv' adalah sebagai berikut:

- Rata-rata Prediksi Volume Pemesanan: 68.71 ribu barel (dibulatkan).

Ini adalah nilai rata-rata dari prediksi volume pemesanan yang dihasilkan oleh model untuk data validasi.

- RMSE (Root Mean Squared Error): 0.89 ribu barel.

RMSE mengukur seberapa baik model kita cocok dengan data yang diamati. Nilai RMSE yang lebih rendah menunjukkan bahwa model memiliki kinerja yang lebih baik.
Dalam konteks ini, RMSE yang sangat rendah menunjukkan bahwa model memiliki kinerja yang sangat baik dalam memprediksi volume cadangan di sumur minyak baru.

- Analisis:

1. Rata-rata prediksi volume pemesanan sangat dekat dengan nilai sebenarnya dari data validasi, yang menunjukkan bahwa model memiliki kinerja yang sangat baik dalam memprediksi volume cadangan.
2. RMSE yang sangat rendah menunjukkan bahwa model memiliki akurasi yang tinggi dalam memprediksi volume cadangan di sumur minyak baru.
3. Dengan demikian, model regresi linear tampaknya sangat cocok untuk dataset 'geo_data_1.csv' dan memberikan hasil yang sangat baik dalam memprediksi volume cadangan.

Dengan demikian, hasil tersebut menunjukkan bahwa model telah berhasil dalam memprediksi volume cadangan di wilayah 'geo_data_1.csv' dengan akurasi yang tinggi. Ini adalah hasil yang sangat positif dan menunjukkan bahwa model dapat dipertimbangkan untuk digunakan dalam pengambilan keputusan terkait pengembangan sumur minyak di wilayah tersebut.

In [7]:
# Latih dan uji model untuk setiap wilayah di 'geo_data_2.csv':

# Pisahkan fitur dan target
X = geo_data_2[['f0', 'f1', 'f2']]
y = geo_data_2['product']

# Pisahkan data menjadi training set dan validation set
X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.25, random_state=42)

# Inisialisasi model regresi linear
model = LinearRegression()

# Latih model menggunakan data training
model.fit(X_train, y_train)

# Buat prediksi untuk data validation
predictions = model.predict(X_val)

# Simpan prediksi dan jawaban yang benar untuk validation set
results_df = pd.DataFrame({'Actual': y_val, 'Predicted': predictions})

# Tampilkan prediksi volume rata-rata pemesanan
print("Rata-rata Prediksi Volume Pemesanan:", predictions.mean())

# Hitung RMSE (Root Mean Squared Error)
rmse = np.sqrt(mean_squared_error(y_val, predictions))
print("RMSE dari Model:", rmse)

Rata-rata Prediksi Volume Pemesanan: 94.77102387765939
RMSE dari Model: 40.14587231134218


Hasil dari pelatihan dan pengujian model untuk wilayah di 'geo_data_2.csv' adalah sebagai berikut:

- Rata-rata Prediksi Volume Pemesanan: 94.77 ribu barel (dibulatkan).

Ini adalah nilai rata-rata dari prediksi volume pemesanan yang dihasilkan oleh model untuk data validasi.

- RMSE (Root Mean Squared Error): 40.15 ribu barel.

RMSE mengukur seberapa baik model kita cocok dengan data yang diamati. Nilai RMSE yang lebih rendah menunjukkan bahwa model memiliki kinerja yang lebih baik.
Dalam konteks ini, RMSE yang cukup besar menunjukkan bahwa model mungkin tidak sesuai dengan data sebaik yang diharapkan.

- Analisis:

1. Rata-rata prediksi volume pemesanan cukup dekat dengan nilai sebenarnya dari data validasi, yang menunjukkan bahwa model mungkin cukup baik dalam melakukan prediksi secara keseluruhan.
2. Namun, RMSE yang cukup tinggi (lebih dari 42% dari rata-rata nilai target) menunjukkan bahwa model mungkin tidak cukup akurat dalam memprediksi volume cadangan di sumur minyak baru.
3. Kemungkinan terdapat faktor-faktor lain yang tidak dimasukkan ke dalam model dan memengaruhi hasil prediksi, atau model yang digunakan mungkin perlu penyesuaian lebih lanjut untuk meningkatkan kinerjanya.

Dengan demikian, hasil tersebut memberikan wawasan awal tentang kinerja model untuk wilayah di 'geo_data_2.csv', namun perlu dilakukan analisis lebih lanjut dan penyesuaian model jika diperlukan untuk meningkatkan akurasinya.










Untuk melakukan perhitungan laba, kita perlu menyimpan semua nilai yang diperlukan dalam variabel terpisah. Nilai-nilai tersebut termasuk:

Investasi awal untuk pengembangan 200 sumur minyak, yaitu $ 100 juta.

Jumlah sumur minyak yang akan dikembangkan, yaitu 200.

Nilai minimal yang harus dihasilkan oleh setiap sumur minyak agar menghindari kerugian, yaitu $500 ribu atau setara dengan 111.1 unit.

Pendapatan yang dihasilkan dari penjualan satu barel minyak mentah, yaitu $ 4.5.

Total pendapatan yang diperoleh dari volume pemesanan yang diterima.

In [8]:
# Investasi awal untuk pengembangan 200 sumur minyak (dalam juta dollar)
investasi_awal = 100

# Jumlah sumur minyak yang akan dikembangkan
jumlah_sumur = 200

# Nilai minimal yang harus dihasilkan oleh setiap sumur minyak untuk menghindari kerugian (dalam ribu dollar)
nilai_minimal = 500

# Pendapatan yang dihasilkan dari penjualan satu barel minyak mentah (dollar per barel)
pendapatan_per_barel = 4.5

# Total pendapatan yang diperoleh dari volume pemesanan yang diterima (dalam ribu dollar)
total_pendapatan = predictions.mean() * pendapatan_per_barel * jumlah_sumur

# Cetak nilai-nilai yang telah disimpan
print("Investasi Awal:", investasi_awal, "juta dollar")
print("Jumlah Sumur Minyak:", jumlah_sumur)
print("Nilai Minimal per Sumur:", nilai_minimal, "ribu dollar")
print("Pendapatan per Barel:", pendapatan_per_barel, "dollar")
print("Total Pendapatan dari Volume Pemesanan:", total_pendapatan, "ribu dollar")

Investasi Awal: 100 juta dollar
Jumlah Sumur Minyak: 200
Nilai Minimal per Sumur: 500 ribu dollar
Pendapatan per Barel: 4.5 dollar
Total Pendapatan dari Volume Pemesanan: 85293.92148989346 ribu dollar




Investasi awal sebesar  100 juta dollar telah menghasilkan pendapatan sebesar 85,293.92 ribu dollar dari volume pemesanan yang diprediksi dari 200 sumur minyak. Namun, pendapatan ini perlu dibandingkan dengan investasi awal dan nilai minimal yang harus dihasilkan oleh setiap sumur minyak.

Meskipun pendapatan tampak cukup tinggi, perlu diperhatikan apakah pendapatan ini sudah mencukupi untuk mengembalikan investasi awal dan memenuhi nilai minimal yang harus dihasilkan oleh setiap sumur minyak.

Analisis lebih lanjut dan perbandingan dengan nilai-nilai lainnya diperlukan untuk menentukan apakah proyek ini menguntungkan secara finansial dan sesuai dengan tujuan bisnis perusahaan.

In [9]:
# Hitung nilai minimal yang harus dihasilkan oleh setiap sumur minyak untuk menghindari kerugian
nilai_minimal_per_sumur = investasi_awal / jumlah_sumur  # dalam juta dollar

# Bandingkan nilai minimal dengan jumlah rata-rata pemesanan di setiap wilayah
print("Nilai Minimal per Sumur:", nilai_minimal_per_sumur, "juta dollar")
print("Jumlah Rata-rata Pemesanan di Geo Data 0:", geo_data_0['product'].mean(), "ribu barel")
print("Jumlah Rata-rata Pemesanan di Geo Data 1:", geo_data_1['product'].mean(), "ribu barel")
print("Jumlah Rata-rata Pemesanan di Geo Data 2:", geo_data_2['product'].mean(), "ribu barel")

Nilai Minimal per Sumur: 0.5 juta dollar
Jumlah Rata-rata Pemesanan di Geo Data 0: 92.50000000000001 ribu barel
Jumlah Rata-rata Pemesanan di Geo Data 1: 68.82500000000002 ribu barel
Jumlah Rata-rata Pemesanan di Geo Data 2: 95.00000000000004 ribu barel


Dari hasil yang telah dihitung:

1. Nilai Minimal per Sumur: $0.5 juta dollar.

- Ini adalah nilai minimal yang harus dihasilkan oleh setiap sumur minyak untuk menghindari kerugian, setara dengan $500 ribu dollar.

2. Jumlah Rata-rata Pemesanan di Geo Data 0: 92.5 ribu barel.

- Ini adalah jumlah rata-rata pemesanan di Geo Data 0.

3. Jumlah Rata-rata Pemesanan di Geo Data 1: 68.825 ribu barel.

- Ini adalah jumlah rata-rata pemesanan di Geo Data 1.

4. Jumlah Rata-rata Pemesanan di Geo Data 2: 95.0 ribu barel.

- Ini adalah jumlah rata-rata pemesanan di Geo Data 2.

Kesimpulan:

Nilai minimal per sumur untuk menghindari kerugian adalah $0.5 juta dollar, sedangkan rata-rata pemesanan di setiap wilayah adalah 92.5 ribu barel untuk Geo Data 0, 68.825 ribu barel untuk Geo Data 1, dan 95.0 ribu barel untuk Geo Data 2.

Dari perbandingan ini, kita dapat melihat bahwa rata-rata pemesanan di Geo Data 0 dan Geo Data 2 lebih tinggi daripada nilai minimal per sumur, sedangkan rata-rata pemesanan di Geo Data 1 lebih rendah. Oleh karena itu, Geo Data 0 dan Geo Data 2 mungkin lebih menguntungkan dari segi pendapatan potensial per sumur dibandingkan dengan Geo Data 1.

Analisis lebih lanjut dan perhitungan lebih detail diperlukan untuk menentukan wilayah mana yang paling cocok untuk pengembangan sumur minyak, mempertimbangkan aspek finansial dan risiko.

In [10]:
predictions_geo_data_0 = model.predict(X_val)  
predictions_geo_data_1 = model.predict(X_val)  
predictions_geo_data_2 = model.predict(X_val)  

def hitung_keuntungan(geo_data, predictions, jumlah_sumur_terpilih=200):
    # Menyortir sumur berdasarkan prediksi tertinggi
    sorted_sumur = geo_data.sort_values(by='product', ascending=False).head(jumlah_sumur_terpilih)
    
    # Menghitung total volume pemesanan dari sumur terpilih
    total_volume_pemesanan = sorted_sumur['product'].sum()
    
    # Menghitung total pendapatan dari volume pemesanan
    total_pendapatan = total_volume_pemesanan * pendapatan_per_barel
    
    # Menghitung total biaya pengembangan sumur
    total_biaya = investasi_awal
    
    # Menghitung keuntungan
    keuntungan = total_pendapatan - total_biaya
    
    return keuntungan

# penggunaan fungsi untuk menghitung keuntungan dari 200 sumur terpilih di Geo Data 0
keuntungan_geo_data_0 = hitung_keuntungan(geo_data_0, predictions_geo_data_0)
print("Keuntungan dari Geo Data 0:", keuntungan_geo_data_0)

# penggunaan fungsi untuk menghitung keuntungan dari 200 sumur terpilih di Geo Data 1
keuntungan_geo_data_1 = hitung_keuntungan(geo_data_1, predictions_geo_data_1)
print("Keuntungan dari Geo Data 1:", keuntungan_geo_data_1)

# penggunaan fungsi untuk menghitung keuntungan dari 200 sumur terpilih di Geo Data 2
keuntungan_geo_data_2 = hitung_keuntungan(geo_data_2, predictions_geo_data_2)
print("Keuntungan dari Geo Data 2:", keuntungan_geo_data_2)

Keuntungan dari Geo Data 0: 166250.36568082406
Keuntungan dari Geo Data 1: 124050.86696681511
Keuntungan dari Geo Data 2: 170496.32928358982


In [11]:
# Mengambil 200 sumur teratas dari setiap wilayah
top_200_geo_data_0 = geo_data_0.sort_values(by='product', ascending=False).head(200)
top_200_geo_data_1 = geo_data_1.sort_values(by='product', ascending=False).head(200)
top_200_geo_data_2 = geo_data_2.sort_values(by='product', ascending=False).head(200)

# Menghitung total volume pemesanan yang diharapkan dari sumur-sumur teratas
total_volume_geo_data_0 = top_200_geo_data_0['product'].sum()
total_volume_geo_data_1 = top_200_geo_data_1['product'].sum()
total_volume_geo_data_2 = top_200_geo_data_2['product'].sum()

# Menampilkan rangkuman
print("Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 0:", total_volume_geo_data_0)
print("Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 1:", total_volume_geo_data_1)
print("Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 2:", total_volume_geo_data_2)

Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 0: 36966.747929072015
Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 1: 27589.081548181137
Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 2: 37910.29539635329


Berdasarkan total volume pemesanan dari 200 sumur teratas di masing-masing wilayah, dapat disimpulkan bahwa:

1. Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 0: 36,966.75 ribu barel
2. otal Volume Pemesanan dari 200 Sumur Teratas di Geo Data 1: 27,589.08 ribu barel
3. Total Volume Pemesanan dari 200 Sumur Teratas di Geo Data 2: 37,910.30 ribu barel

Dari hasil di atas, wilayah yang paling cocok untuk pengembangan sumur minyak adalah Geo Data 2 karena memiliki total volume pemesanan yang paling tinggi di antara ketiga wilayah tersebut. Total volume pemesanan yang tinggi menandakan adanya potensi produksi minyak yang besar, yang dapat menghasilkan keuntungan yang lebih besar juga.

In [12]:
# Hitung keuntungan dari 200 sumur terpilih di Geo Data 2
keuntungan_geo_data_2 = hitung_keuntungan(top_200_geo_data_2, predictions_geo_data_2)
print("Keuntungan dari Geo Data 2:", keuntungan_geo_data_2)

Keuntungan dari Geo Data 2: 170496.32928358982


Dengan demikian, Geo Data 2 direkomendasikan untuk pengembangan sumur minyak karena memiliki potensi volume pemesanan yang tinggi, yang kemudian dapat menghasilkan keuntungan yang signifikan.

In [13]:
def bootstrapping_profit(predictions, jumlah_sumur_terpilih=200, jumlah_sampel=1000):
    # Mengambil prediksi untuk 200 sumur teratas
    top_200_predictions = predictions[:jumlah_sumur_terpilih]
    
    # Menghitung profit dari 200 sumur teratas
    profit = top_200_predictions.sum() * pendapatan_per_barel - investasi_awal
    
    # Bootstrapping untuk menghitung distribusi profit
    profits = []
    for _ in range(jumlah_sampel):
        # Mengambil sampel dengan pengembalian
        sample = np.random.choice(top_200_predictions, size=jumlah_sumur_terpilih, replace=True)
        # Menghitung profit dari sampel
        sample_profit = sample.sum() * pendapatan_per_barel - investasi_awal
        profits.append(sample_profit)
    
    # Menghitung statistik dari distribusi profit
    mean_profit = np.mean(profits)
    std_profit = np.std(profits)
    confidence_interval = np.percentile(profits, [2.5, 97.5])
    
    return mean_profit, std_profit, confidence_interval

# Bootstrapping untuk Geo Data 0
mean_profit_geo_data_0, std_profit_geo_data_0, confidence_interval_geo_data_0 = bootstrapping_profit(predictions_geo_data_0)

# Bootstrapping untuk Geo Data 1
mean_profit_geo_data_1, std_profit_geo_data_1, confidence_interval_geo_data_1 = bootstrapping_profit(predictions_geo_data_1)

# Bootstrapping untuk Geo Data 2
mean_profit_geo_data_2, std_profit_geo_data_2, confidence_interval_geo_data_2 = bootstrapping_profit(predictions_geo_data_2)

# Menampilkan hasil
print("Geo Data 0:")
print("Rata-rata Profit:", mean_profit_geo_data_0)
print("Standar Deviasi Profit:", std_profit_geo_data_0)
print("Interval Kepercayaan 95%:", confidence_interval_geo_data_0)

print("\nGeo Data 1:")
print("Rata-rata Profit:", mean_profit_geo_data_1)
print("Standar Deviasi Profit:", std_profit_geo_data_1)
print("Interval Kepercayaan 95%:", confidence_interval_geo_data_1)

print("\nGeo Data 2:")
print("Rata-rata Profit:", mean_profit_geo_data_2)
print("Standar Deviasi Profit:", std_profit_geo_data_2)
print("Interval Kepercayaan 95%:", confidence_interval_geo_data_2)

Geo Data 0:
Rata-rata Profit: 84329.47941728705
Standar Deviasi Profit: 1204.9688078004338
Interval Kepercayaan 95%: [81983.59477499 86664.44980245]

Geo Data 1:
Rata-rata Profit: 84270.89534088972
Standar Deviasi Profit: 1172.3251437576534
Interval Kepercayaan 95%: [81925.96955024 86458.0914883 ]

Geo Data 2:
Rata-rata Profit: 84308.13185272139
Standar Deviasi Profit: 1194.3715806466523
Interval Kepercayaan 95%: [81979.1875595 86632.8978147]


In [14]:
def calculate_bootstrap_profit(predictions, n_samples=1000, n_wells=200):
    profits = []
    for _ in range(n_samples):
        # Sampling dengan pengembalian dari prediksi
        sampled_predictions = np.random.choice(predictions, size=n_wells, replace=True)
        # Menghitung total pendapatan dari volume pemesanan
        total_pendapatan = np.sum(sampled_predictions)
        # Menghitung total biaya pengembangan sumur
        total_biaya = investasi_awal
        # Menghitung keuntungan
        keuntungan = total_pendapatan - total_biaya
        profits.append(keuntungan)
    return profits

# Bootstrapping untuk masing-masing wilayah
profits_geo_data_0 = calculate_bootstrap_profit(predictions_geo_data_0)
profits_geo_data_1 = calculate_bootstrap_profit(predictions_geo_data_1)
profits_geo_data_2 = calculate_bootstrap_profit(predictions_geo_data_2)

In [15]:
def calculate_loss_risk(profits):
    # Menghitung jumlah sampel yang menghasilkan profit negatif
    negative_profits = np.array([profit for profit in profits if profit < 0])
    num_negative_profits = len(negative_profits)
    
    # Menghitung risiko kerugian sebagai persentase
    loss_risk = (num_negative_profits / len(profits)) * 100
    
    return loss_risk

# Menghitung risiko kerugian dan statistik lainnya untuk masing-masing wilayah
loss_risk_geo_data_0 = calculate_loss_risk(profits_geo_data_0)
loss_risk_geo_data_1 = calculate_loss_risk(profits_geo_data_1)
loss_risk_geo_data_2 = calculate_loss_risk(profits_geo_data_2)

# Menampilkan hasil
print("Geo Data 0:")
print("Rata-rata Profit:", mean_profit_geo_data_0)
print("Interval Kepercayaan 95%:", confidence_interval_geo_data_0)
print("Risiko Kerugian:", loss_risk_geo_data_0, "%")

print("\nGeo Data 1:")
print("Rata-rata Profit:", mean_profit_geo_data_1)
print("Interval Kepercayaan 95%:", confidence_interval_geo_data_1)
print("Risiko Kerugian:", loss_risk_geo_data_1, "%")

print("\nGeo Data 2:")
print("Rata-rata Profit:", mean_profit_geo_data_2)
print("Interval Kepercayaan 95%:", confidence_interval_geo_data_2)
print("Risiko Kerugian:", loss_risk_geo_data_2, "%")

Geo Data 0:
Rata-rata Profit: 84329.47941728705
Interval Kepercayaan 95%: [81983.59477499 86664.44980245]
Risiko Kerugian: 0.0 %

Geo Data 1:
Rata-rata Profit: 84270.89534088972
Interval Kepercayaan 95%: [81925.96955024 86458.0914883 ]
Risiko Kerugian: 0.0 %

Geo Data 2:
Rata-rata Profit: 84308.13185272139
Interval Kepercayaan 95%: [81979.1875595 86632.8978147]
Risiko Kerugian: 0.0 %


Berdasarkan hasil yang diperoleh, kesimpulan yang dapat diambil adalah:

- Ketiga wilayah memiliki rata-rata profit yang hampir sama. 
- Dengan interval kepercayaan 95% yang juga saling tumpang tindih.
- Risiko kerugian untuk setiap wilayah adalah 0%, yang berarti tidak ada probabilitas keuntungan menjadi negatif.

Dari kesimpulan ini, tidak ada perbedaan signifikan dalam rata-rata profit dan risiko kerugian antara ketiga wilayah. Oleh karena itu, tidak ada wilayah yang secara signifikan lebih baik atau lebih buruk untuk pengembangan sumur minyak daripada yang lain.

Kesimpulan ini tidak sesuai dengan pilihan sebelumnya di poin 4.3, di mana kita merekomendasikan wilayah dengan keuntungan tertinggi. Ini menunjukkan bahwa keputusan awal didasarkan pada prediksi model dan keuntungan tanpa mempertimbangkan variasi dan risiko yang mungkin terjadi dalam praktiknya. Dalam situasi sebenarnya, akan lebih bijaksana untuk mempertimbangkan variasi dan risiko lebih lanjut sebelum membuat keputusan pengembangan.

# Kesimpulan umum :

Dalam proyek ini, kita telah melakukan analisis dan pengembangan model untuk mengevaluasi potensi pengembangan sumur minyak di tiga wilayah yang berbeda. Langkah-langkah yang dilakukan meliputi:

1. Eksplorasi data: Mengunduh, memuat, dan menjelajahi data untuk memahami karakteristiknya.
2. Pembangunan model: Melatih model regresi linear untuk memprediksi volume pemesanan minyak dari fitur-fitur geologis yang tersedia.
3. Evaluasi model: Mengukur kinerja model menggunakan metrik seperti RMSE dan menganalisis hasil prediksi untuk masing-masing wilayah.
4. Perhitungan laba dan risiko: Menghitung keuntungan yang diharapkan dari pengembangan sumur minyak berdasarkan prediksi model, dan mengevaluasi risiko kerugian menggunakan teknik bootstrapping.

Kesimpulan umum dari proyek ini adalah sebagai berikut:

- Meskipun model regresi linear memberikan perkiraan volume pemesanan yang cukup baik, faktor-faktor lain seperti variasi alamiah dalam kualitas dan kuantitas minyak serta faktor eksternal seperti fluktuasi harga minyak juga perlu dipertimbangkan dalam pengambilan keputusan pengembangan.
- Meskipun risiko kerugian dalam pengembangan sumur minyak diperkirakan rendah, tetapi perlu diingat bahwa ini hanya estimasi berdasarkan prediksi model. Faktor-faktor lain seperti ketidakpastian geologi, biaya produksi, dan perubahan regulasi juga dapat berdampak pada risiko sebenarnya.
- Dalam pengambilan keputusan pengembangan, penting untuk melakukan analisis yang komprehensif dan mempertimbangkan semua faktor yang relevan, bukan hanya hasil dari model prediksi saja.

Secara keseluruhan, proyek ini menyoroti pentingnya penggunaan analisis data dan pemodelan yang cermat dalam pengambilan keputusan bisnis, terutama dalam industri yang sangat bergantung pada faktor-faktor geologis dan ekonomi yang kompleks seperti industri perminyakan.
